.github/workflows/sprocket-test.yml

```
@@ -0,0 +1,143 @@
+          {
+            "total": $total,
+            "passed": $passed,
+            "warnings": $warnings,
+            "failures": $failed,
+            "invalid_outputs": $invalid,
+            "ignored": $ignored
+          }
+          EOF
+
+          # Copy list of failures to artifacts
+          cp sprocket_test/failed_tests.txt sprocket_tests/artifacts/failed_tests.txt
+
+          # Write JUnit XML (for GitHub UI annotations)
+          cat <<EOF > sprocket_tests/artifacts/results.xml
+          <?xml version="1.0" encoding="UTF-8"?>
+          <testsuite name="Sprocket WDL 1.2 Conformance" tests="$total" failures="$failed" skipped="$ignored">
+            <testcase classname="Sprocket" name="wdl-1.2-conformance">
+              $(if [ "$failed" -gt 0 ]; then echo "<failure message=\"Failures=$failed, Invalid=$invalid\"/>"; fi)
+            </testcase>
+          </testsuite>
+          EOF
+
+          # Generate shields.io badge JSON
+          color="green"
+          if [ "$passed" -lt "$total" ]; then color="yellow"; fi
+          if [ "$passed" -eq 0 ]; then color="red"; fi
+          cat <<EOF > sprocket_tests/artifacts/shields.json
+          {
+            "schemaVersion": 1,
+            "label": "Sprocket WDL 1.2",
+            "message": "$passed/$total passed",
+            "color": "$color"
+          }
+          EOF
+
+      - name: Upload test artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: sprocket-test-results
+          path: sprocket_tests/artifacts/
+
+      - name: Publish badge JSON to current branch
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+          BRANCH="${GITHUB_REF_NAME}"
+          REPO_URL="https://x-access-token:${GITHUB_TOKEN}@github.com/${{ github.repository }}.git"
+          
+          git fetch origin $BRANCH
+          git checkout $BRANCH
+          
+          mkdir -p shields
+          cp sprocket_tests/artifacts/shields.json shields/sprocket_shields.json
+          git add shields/sprocket_shields.json
+
+          # Update README badge link only if needed
+          sed_pattern="https://raw.githubusercontent.com/openwdl/wdl/.*/shields/sprocket_shields.json"
+          new_url="https://raw.githubusercontent.com/openwdl/wdl/$BRANCH/shields/sprocket_shields.json"
+          if grep -qE "$sed_pattern" README.md; then
+            git pull
+            sed -E -i "/sprocket_shields\.json/s|$sed_pattern|$new_url|g" README.md
+            sed -E -i "/sprocket_shields\.json/s|$run_pattern|$new_run|g" README.md
+            git add README.md
+          fi
+
+          # Commit only if there are staged changes
+          if ! git diff --cached --quiet; then
+            git commit -m "Update Sprocket shields badge for $BRANCH"
```

```suggestion
            git commit -m "chore: updates Sprocket shields badge for `$BRANCH`"
```

Same comment as above.

---

agreed